data_filter_mongodb/internal/mongo/mongo.go

```diff
@@ -0,0 +1,144 @@
+package mongo
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"go.uber.org/zap"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type DBClient struct {
+	Address string
+	Client  *mongo.Client
+}
+
+type MongoClientStruct struct {
+	Mongo         *mongo.Client
+	Database      string
+	ClientOptions *options.ClientOptions
+	Logger        *zap.Logger
+}
+
+type MongoClient interface {
+	QueryMongo(pipeline []Queries) Result
+	GetAllDataFromMongo(w http.ResponseWriter)
+}
+
+// Result contains partially evaluating OPA queries.
+type Result struct {
+	Defined bool
+	Data    []EmployeeDetails
+}
+
+type Queries struct {
+	Pipeline bson.M
+}
+
+func (c *MongoClientStruct) CreateConnection() *mongo.Client {
+	client, err := mongo.Connect(context.Background(), c.ClientOptions)
+	if err != nil {
+		c.Logger.Error("Unable to establish connection with mongo client",
+			zap.String("uri", c.ClientOptions.GetURI()), zap.Error(err))
+	}
+
+	err = client.Ping(context.TODO(), nil)
+	if err != nil {
+		c.Logger.Error("Unable to ping the mongo db ", zap.Error(err))
+	}
+	return client
+}
+
+func (c *MongoClientStruct) QueryMongo(pipeline []Queries) Result {
+	var result Result
+	var queries []bson.M
+	for _, q := range pipeline {
+		queries = append(queries, q.Pipeline)
+	}
+	finalQuery := bson.M{"$or": queries}
+	c.Logger.Info("mongo query: ", zap.Any("", finalQuery))
+	collection := c.Mongo.Database(c.Database).Collection("employees")
+	f, err := collection.Find(context.TODO(), finalQuery)
+	if err != nil {
+		c.Logger.Error("Failed to query mongo database", zap.Error(err))
+	}
+
+	for f.Next(context.Background()) {
+		e1 := &EmployeeDetails{}
+		err = f.Decode(&e1)
+		if err != nil {
+			c.Logger.Error("Failed to decode the row queried from db.", zap.Error(err))
+		}
+		result.Data = append(result.Data, *e1)
+	}
+
+	result.Defined = true
+	return result
+}
+
+func (c *MongoClientStruct) GetAllDataFromMongo(w http.ResponseWriter) {
+	var records []EmployeeDetails
+	f, err := c.Mongo.Database(c.Database).Collection("employees").Find(context.Background(), bson.M{})
+	if err != nil {
+		c.Logger.Error("Failed to query mongo database", zap.Error(err))
+	}
+
+	for f.Next(context.Background()) {
+		e1 := &EmployeeDetails{}
+		err = f.Decode(&e1)
+		if err != nil {
+			c.Logger.Error("Failed to decode the row queried from db.", zap.Error(err))
+		}
+		records = append(records, *e1)
+	}
+	resultBytes, err := json.MarshalIndent(records, "", "  ")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	_, err = w.Write(resultBytes)
+	if err != nil {
+		c.Logger.Error("Failed to write received result to response writer.", zap.Error(err))
+	}
+}
+
```

Comment:
It would be helpful to add a section on `Generated Mongo queries` in the README.